vertx-db2-client/src/test/java/io/vertx/db2client/DB2ErrorMessageTest.java

```diff
@@ -0,0 +1,91 @@
+		  ctx.assertTrue(err instanceof DB2Exception, "The error message returned is of the wrong type.  It should be a DB2Exception, but it was of type " + err.getClass().getSimpleName());
+//		  DB2Exception ex = (DB2Exception) err;
+//		  ctx.assertTrue(ex.getMessage().contains("Invalid database"), "The SQL error message returned is not correct.  It should have contained \"Invalid database\", but instead it said \"" + ex.getMessage() + "\"");
+	  }));
+	}
+	
+	@Test
+	public void testConnectInvalidUsername(TestContext ctx) {
+	  options.setUser("INVALID_USER_FOR_TESTING");
+	  DB2Connection.connect(vertx, options, ctx.asyncAssertFailure(err -> {
+		  ctx.assertTrue(err instanceof DB2Exception, "The error message returned is of the wrong type.  It should be a DB2Exception, but it was of type " + err.getClass().getSimpleName());
+		  DB2Exception ex = (DB2Exception) err;
+		  ctx.assertTrue(ex.getMessage().contains("Invalid credentials"), "The SQL error message returned is not correct.  It should have contained \"Invalid credentials\", but instead it said \"" + ex.getMessage() + "\"");
+		  ctx.assertTrue(ex.getErrorCode() == SqlCode.INVALID_CREDENTIALS, "The SQL error code returned is not correct.  It should have been " + SqlCode.INVALID_CREDENTIALS + ", but it was " + ex.getErrorCode());
+		  ctx.assertTrue(ex.getSqlState() == SQLState.NET_CONNECT_AUTH_FAILED, "The SQL state returned is not correct.  It should have been " + SQLState.NET_CONNECT_AUTH_FAILED + ", but it was " + ex.getSqlState());
+		  System.out.println(err.getMessage());
+	  }));
+	}
+	
+	@Test
+	public void testConnectInvalidPassword(TestContext ctx) {
+		options.setPassword("INVALID_PASSWORD_FOR_TESTING");
+		  DB2Connection.connect(vertx, options, ctx.asyncAssertFailure(err -> {
+			  ctx.assertTrue(err instanceof DB2Exception, "The error message returned is of the wrong type.  It should be a DB2Exception, but it was of type " + err.getClass().getSimpleName());
+			  DB2Exception ex = (DB2Exception) err;
+			  ctx.assertTrue(ex.getMessage().contains("Invalid credentials"), "The SQL error message returned is not correct.  It should have contained \"Invalid credentials\", but instead it said \"" + ex.getMessage() + "\"");
+			  ctx.assertTrue(ex.getErrorCode() == SqlCode.INVALID_CREDENTIALS, "The SQL error code returned is not correct.  It should have been " + SqlCode.INVALID_CREDENTIALS + ", but it was " + ex.getErrorCode());
+			  ctx.assertTrue(ex.getSqlState() == SQLState.NET_CONNECT_AUTH_FAILED, "The SQL state returned is not correct.  It should have been " + SQLState.NET_CONNECT_AUTH_FAILED + ", but it was " + ex.getSqlState());
+			  System.out.println(err.getMessage());
+		  }));
+	}
+	
+	@Test
+	@Ignore  // TODO - @GJW I want to validate we would always want to fail a test for a blank user name.  If so, we want to catch this *before* we connect to the DB2 server
+	public void testConnectBlankUsername(TestContext ctx) {
+	  options.setUser("");
```

Comment:
If we were to validate early in the blank username/password case, we would have it blow up immediately when `setUser/setPassword` is called, instead of waiting until the user tries to connect with the `options`

Reply:
I will handle that in a future PR.